liquibase-core/src/main/java/liquibase/resource/ClassLoaderResourceAccessor.java

```diff
@@ -72,24 +72,29 @@ public ClassLoaderResourceAccessor(ClassLoader classLoader) {
             URL fileUrl = fileUrls.nextElement();
 
             if (fileUrl.toExternalForm().startsWith("jar:file:")
-                    || fileUrl.toExternalForm().startsWith("wsjar:file:")
-                    || fileUrl.toExternalForm().startsWith("zip:")) {
+                || fileUrl.toExternalForm().startsWith("wsjar:file:")
+                || fileUrl.toExternalForm().startsWith("zip:")) {
 
                 String[] zipAndFile = fileUrl.getFile().split("!");
                 String zipFilePath = zipAndFile[0];
+                path = zipAndFile[1];
```

Comment:
this won't work for implementations like spring-boot. There you have one fat-jar which contains other jars inside -> spring-boot-fat.jar!/BOOT-INF/libs/implementation-lib.jar!/db/change-log/change-log.xml